### PR TITLE
docs: new doc structure

### PR DESCRIPTION
**Pull request by ntarocco**
Invenio_Framework_v3_—_Invenio_3_4_0a6_documentation

For reviewers:
- checkout the PR
- `venv` + `pip install -e .[docs]`
- `sphinx-build -qnNW docs docs/_build/html`
- `open docs/_build/html/index.html`

---
closes https://github.com/inveniosoftware/invenio/issues/4035.

### CODE REVIEW

File: docs/documentation/how-to/build-a-module.rst [hunk 1]
```diff
@@ -164,7 +164,7 @@ ext.py
 >>>>>>
```
Comment (topless) [inline docs/documentation/how-to/build-a-module.rst:164]: Are these actual characters or merge leftovers?

File: docs/documentation/main-concepts/repository-structure.rst [hunk 2]
```diff
@@ -7,8 +7,10 @@
 
 .. _build-repository:
 
-Build a repository
-==================
+Repository structure
+====================
+
+CHANGE ME PLEASE
```
Comment (topless) [inline docs/documentation/main-concepts/repository-structure.rst:13]: Maybe a `TODO` comment with what to change, or the actual content even better

**Comment by diegodelemos** [comment]
Needs rebase and a few amendments cc @FlorianCassayre

**Comment by FlorianCassayre** [comment]
Thanks for the review, will fix.

File: docs/documentation/main-concepts/repository-structure.rst (hunk 2 above)
Comment (FlorianCassayre) [reply docs/documentation/main-concepts/repository-structure.rst]: Forgot about that, thanks for pointing it out. It was to remind us that the tone of the page did not match the rest; I updated it.

File: docs/documentation/how-to/build-a-module.rst (hunk 1 above)
Comment (FlorianCassayre) [reply docs/documentation/how-to/build-a-module.rst:164]: (those are subtitles)

File: docs/documentation/bundles/index.rst [hunk 3]
```diff
@@ -1,8 +1,16 @@
 ..
     This file is part of Invenio.
-    Copyright (C) 2017-2018 CERN.
+    Copyright (C) 2018-2020 CERN.
```
Comment (ppanero) [inline docs/documentation/bundles/index.rst:3]: minor: if the start was 2017, shouldn't it be kept?

File: docs/documentation/how-to/index.rst [hunk 4]
```diff
@@ -1,10 +1,15 @@
 ..
     This file is part of Invenio.
-    Copyright (C) 2017-2018 CERN.
+    Copyright (C) 2018-2020 CERN.
```
Comment (ppanero) [inline docs/documentation/how-to/index.rst:3]: minor: start year?

**Review by ppanero — COMMENTED** [review]
Note, this comments might not be fully about structure itself (sorry about that), they just came out when reviewing. So feel free to open issues and shelve them if not applicable.

- In `getting-help` we mention that maintainers are set in `MAINTAINERS` file, are those actually maintained? Or should we simply point to the `opensource/repositories.yaml`? e.g. I never put myself there in any repo... Maybe I should have, I never noticed the file/didn't know I had to.
- In the *Developer environment guide* we point to `pyenv` for python. Should we add a section on JS? (e.g. `nvm`)?

**Comment by diegodelemos** [comment]
> In getting-help we mention that maintainers are set in MAINTAINERS file, are those actually maintained? Or should we simply point to the opensource/repositories.yaml? e.g. I never put myself there in any repo... Maybe I should have, I never noticed the file/didn't know I had to.

This makes complete sense, however is not to be handled in this PR, it is exactly what one of the cards we have left in the sprint deals with. How to give better visibility to the maintainer(s) of a module, groups of interest etc... 

image

@lnielsen  shall we speed it up and try to get something nailed down regarding this topic? Otherwise, I would leave it for the future.

> In the Developer environment guide we point to pyenv for python. Should we add a section on JS? (e.g. nvm)?

I think this would be useful, somehow we have forgotten about it while creating/reviewing https://github.com/inveniosoftware/invenio/pull/4041.

**Comment by FlorianCassayre** [comment]
I eventually managed to fix [this](https://github.com/inveniosoftware/invenio/pull/4046#issuecomment-678270799) by override the `navigation.html` template. The overriden code only consists of 10 lines so it's not too bad. This allows to fully expand the sidebar on the selected page, while keeping a shallow homepage (`maxdepth` of `2`).

image

I also included the previous/next buttons at the bottom of the page.

**Review by diegodelemos — APPROVED** [review]

**Pull request merged**